src/qt/bitcoingui.cpp

```diff
@@ -1317,6 +1323,15 @@ void BitcoinGUI::updateProxyIcon()
     }
 }
 
+void BitcoinGUI::updateTorIcon()
+{
+    if (clientModel->hasTorOnlyConnections()) {
+        m_tor_icon->setPixmap(platformStyle->SingleColorIcon(":/icons/tor_connected").pixmap(STATUSBAR_ICONSIZE, STATUSBAR_ICONSIZE));
+        m_tor_icon->setToolTip(tr("All connections are <b>via Tor</b> only"));
+        m_tor_icon->show();
+    }
+}
```

Comment:
Should this have an else case:
```
else {
	m_tor_icon->hide();
}
```

Reply:
Done in https://github.com/bitcoin-core/gui/pull/86.